dashel/dashel-posix.cpp

```diff
@@ -1078,15 +1079,25 @@ namespace Dashel
 				ioctl(fd, TIOCMBIS, &iFlags);
 			else
 				ioctl(fd, TIOCMBIC, &iFlags);
+			
+			struct sigaction act;
+			memset(&act, 0, sizeof (act));
+			act.sa_handler = &SerialStream::handle_sighup;
+			sigaction(SIGHUP, &act, &oldsigaction);
 		}
 		
 		//! Destructor, restore old serial port state
 		virtual ~SerialStream()
 		{
 			 tcsetattr(fd, TCSANOW, &oldtio);
+			 sigaction(SIGHUP, &oldsigaction, 0);
 		}
 		
 		virtual void flush() { /* hook for use by derived classes */ }
+		
+		static void handle_sighup (int ) {
```

Comment:
It could be named handleSighub, but as it is signal processing, I would also accept this naming.